src/lib.rs

```diff
@@ -15,16 +15,19 @@ pub use {error::Error, expr::Expr, geometry::Geometry, parser::parse_text, valid
 /// A SQL query, broken into the query and parameters.
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct SqlQuery {
-    query: String,
-    params: Vec<String>,
+    /// The SQL query, with placeholders for parameters.
+    pub query: String,
+
+    /// The SQL parameters, as strings.
+    pub params: Vec<String>,
```

Comment:
Does this potentially need validation, e.g. that the `query` has the same number of parameters as `params`? If so, maybe it would be better to remove the `pub` fields and add a constructor instead?

Reply:
The to_sql is going to be undergoing a fair amount of change soon, so I wouldn't spend too much time monkeying with anything surrounding it right now.